hydra-node/src/Hydra/Options.hs

```diff
@@ -310,22 +311,24 @@ chainConfigParser =
     <*> contestationPeriodParser
 
 networkIdParser :: Parser NetworkId
-networkIdParser =
-  testnetParser
+networkIdParser = pMainnet <|> fmap Testnet pTestnetMagic
  where
-  testnetParser =
-    Testnet . NetworkMagic
+  pMainnet :: Parser NetworkId
+  pMainnet =
+    flag'
+      Mainnet
+      ( long "mainnet"
+          <> help "Use the mainnet magic id."
+      )
+
+  pTestnetMagic :: Parser NetworkMagic
+  pTestnetMagic =
+    NetworkMagic
       <$> option
         auto
-        ( long "network-id"
-            <> metavar "INTEGER"
-            <> value 42
-            <> showDefault
-            <> completer (listCompleter ["1", "2", "42"])
-            <> help
-              "Network identifier for a testnet to connect to. We only need to \
-              \provide the magic number here. For example: '2' is the 'preview' \
-              \network. See https://book.world.dev.cardano.org/environments.html for available networks."
```

Comment:
This explanation was actually quite nice, also the completer was good.
Furthermore, there was a default for the `42` testnet (which is the local devnet for our demos).

Are we sure we have the same behavior and UX still in the demos and examples if we don't have this default??

Should: Keep the good explanation.